backend/src/api/paths/problems.ts

```diff
@@ -194,28 +226,96 @@ const getSubmissionsByProblemIdRoute = createRoute({
 })
 
 // ルートの設定
-app.openapi(getProblemsRoute, (c) => {
-  // TODO: 実際のデータベースクエリを実装
-  const problems: z.infer<typeof schemas.Problem>[] = [
-    {
-      body: "この問題の本文です。",
-      id: 1,
-      supported_languages: [{ name: "Python", version: "3.9" }],
-      test_cases: [{ input: "入力例", output: "出力例" }],
-      title: "サンプル問題",
+app.openapi(getProblemsRoute, async (c) => {
+  const problems = await prisma.problem.findMany({
+    include: {
+      supportedLanguages: {
+        include: {
+          language: true,
+        },
+      },
+      testCases: true,
     },
-  ]
-  return c.json(problems)
+  })
+
+  const formattedProblems = problems.map((problem: ProblemWithRelations) => ({
+    body: problem.body,
+    id: problem.id,
+    supported_languages: problem.supportedLanguages.map(
+      (supportedLang: ProblemWithRelations["supportedLanguages"][0]) => ({
+        name: supportedLang.language.name,
+        version: supportedLang.language.version,
+      }),
+    ),
+    test_cases: problem.testCases.map(
+      (testCase: ProblemWithRelations["testCases"][0]) => ({
```

Comment:
IMO: problemの型は推論できるので、明示的に書かなくて良いかなと思いました

```suggestion
      (testCase) => ({
```